dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/dataintegrity/DataIntegrityUsersWithAllAuthorityControllerTest.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * @author Jason P. Pickering
+ */
+class DataIntegrityUsersWithAllAuthorityControllerTest
+    extends AbstractDataIntegrityIntegrationTest {
+
+  private static final String CHECK_NAME = "users_with_all_authority";
+
+  private static final String DETAILS_ID_TYPE = "users";
+
+  @Test
+  void testCanIdentifyUsersWithAllAuthority() {
+
+    String orgunitAUid =
+        assertStatus(
+            HttpStatus.CREATED,
+            POST(
+                "/organisationUnits",
+                "{ 'name': 'Fish District', 'shortName': 'Fish District', 'openingDate' : '2022-01-01'}"));
+
+    String userRoleUidA =
+        assertStatus(
+            HttpStatus.CREATED,
+            POST("/userRoles", "{ 'name': 'ALL role', 'authorities': ['ALL'] }"));
+
+    assertStatus(
+        HttpStatus.CREATED,
+        POST(
+            "/users",
+            "{ 'username': 'bobbytables' , 'password': 'District123+', 'firstName': 'Bobby', 'surname': 'Tables', 'organisationUnits' : [{'id' : '"
+                + orgunitAUid
+                + "'}], 'dataViewOrganisationUnits' : [{'id' : '"
+                + orgunitAUid
+                + "'}], 'userRoles' : [{'id' : '"
+                + userRoleUidA
+                + "'}]}"));
+
+    // Note that there is already one user which exists due to the overall test setup, thus, two
+    // users in total.
+    List<User> allUsers = userService.getAllUsers();
+    assertEquals(2, allUsers.size());
+    Set<String> userNames = new HashSet<>();
+    for (User user : allUsers) {
+      userNames.add(user.getUsername());
+    }
+    Set<String> userUids = new HashSet<>();
+    for (User user : allUsers) {
+      userUids.add(user.getUid());
+    }
+    Set<String> userComments = new HashSet<>();
+    // Each user should be active, so just create a set of "Active" for each user
+    allUsers.forEach(user -> userComments.add("Active"));
+
+    // Add a non-ALL authority user
+
+    String userRoleUidB =
+        assertStatus(
+            HttpStatus.CREATED,
+            POST("/userRoles", "{ 'name': 'Not all role', 'authorities': ['F_DATAVALUE_ADD'] }"));
+
+    assertStatus(
+        HttpStatus.CREATED,
+        POST(
+            "/users",
+            "{ 'username': 'bobbytables2' , 'password': 'District123+', 'firstName': 'Bobby', 'surname': 'Tables', 'organisationUnits' : [{'id' : '"
+                + orgunitAUid
+                + "'}], 'dataViewOrganisationUnits' : [{'id' : '"
+                + orgunitAUid
+                + "'}], 'userRoles' : [{'id' : '"
+                + userRoleUidB
+                + "'}]}"));
+
+    // Note the expected percentage is 2/3 = 66%
+    assertHasDataIntegrityIssues(
+        DETAILS_ID_TYPE, CHECK_NAME, 66, userUids, userNames, userComments, true);
+
+    JsonDataIntegrityDetails details = getDetails(CHECK_NAME);
+    JsonList<JsonDataIntegrityDetails.JsonDataIntegrityIssue> issues = details.getIssues();
+    assertEquals(2, issues.size());
```

Comment:
Maybe add another assert for the UIDs we expect on the issues found to make sure